### ISSUE TITLE
[css-writing-modes] spacing within text-combine-horizontal

### ISSUE THREAD

**Issue by fantasai** [w3c/csswg-drafts#9423]
https://www.w3.org/TR/css-writing-modes-4/#text-combine-layout
> When combining text as for text-combine-upright: all, the glyphs of the combined text are bidi-isolated and composed horizontally (ignoring letter-spacing and any forced line breaks, but using the specified font settings), similar to the contents of an inline-block box with a horizontal writing mode and a line-height of 1em.

We specify ignoring `letter-spacing`, but what should we do about `word-spacing`, `text-autospace` and `text-spacing-trim`? Should these be honored or ignored? @kojiishi @MurakamiShinyu what do you think?

**Comment by fantasai** [comment]
Fwiw, my inclination is to ignore all of the spacing properties, and force `text-spacing-trim` to behave like `trim-all`.

**Comment by frivoal** [comment]
I agree the question is worth answering, and I probably agree with @fantasai 's suggestion above, although it could make sense to give authors a choice between `trim-all` and and no trimming at all.

**Comment by frivoal** [comment]
Not sure if this should be considered part of the same issue or an adjacent one, https://wpt.fyi/results/css/css-text/text-autospace/text-autospace-vertical-combine-001.html?label=experimental&label=master&aligned assumes that text-autospace has no effect at the boundaries of a text-combine-upright. I think that can be derived from existing text spec, as autospace
> controls spacing between adjacent characters on the same line within the same inline formatting context

(cf https://drafts.csswg.org/css-text-4/#text-autospace-property)

and 

> When combining text as for text-combine-upright: all, the glyphs of the combined text are bidi-isolated and composed horizontally (ignoring letter-spacing and any forced line breaks, but using the specified font settings), similar to the contents of an inline-block box 

(cf https://www.w3.org/TR/css-writing-modes-3/#text-combine-layout)

so you could say that `text-combine-upright` creates a distinct inline formatting context, and that's why there's no spacing. But it is a bit indirect, and not very precise ("similar to […]"). I think we could try and make that a tad more explicit.

**Comment by fantasai** [comment]
I don't think there's much utility to giving a choice here, and adding more properties and values to do it seems excessive.
